engine/standalone/util/src/main/scala/pl/touk/nussknacker/engine/standalone/utils/typed/SchemaTypingResult.scala

```diff
@@ -0,0 +1,80 @@
+package pl.touk.nussknacker.engine.standalone.utils.typed
+
+import io.circe.Json
+import pl.touk.nussknacker.engine.api.CirceUtil
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedObjectTypingResult, TypingResult}
+
+object SchemaTypingResult {
+
+  final val TYPE_FIELD = "type"
+  final val DEFAULT_FIELD = "default"
+  final val PROPERTIES_FIELD = "properties"
+  final val ARRAY_ITEMS_FIELD = "items"
+
+  /* *
+  * It is simplified schema parser. We define schema as an schema type=object : "{ "properties": { our definition } }"
+  * Rules:
+  * - Any type=object to have "properties" field
+  * - Any type=array to have "items" field
+  * */
+
+  def jsonSchemaToTypingResult(definition: String): TypingResult = {
+    val json = CirceUtil.decodeJsonUnsafe[Json](definition, "Provided json-schema is not valid")
+    parseJsonObject(json)
+  }
+
+  private def parseJsonObject(json: Json): TypingResult = {
+    val properties = json.hcursor.downField(PROPERTIES_FIELD).focus
+    if(properties.isEmpty) {
+      throw new IllegalArgumentException(s"There should be exactly one '$PROPERTIES_FIELD' field in 'object'.")
+    }
+    else jsonValuesToTypingResult(properties.get)
+  }
+
+  private def jsonValuesToTypingResult(json: Json): TypingResult = {
+    if(containsFieldType(json)) resolveJsonTypingResult(json)
+    else {
+      TypedObjectTypingResult(getJsonFirstLevelFieldsMap(json).mapValues(jsonValuesToTypingResult).toList)
+    }
+  }
+
+  private def containsFieldType(json: Json): Boolean = {
+    lazy val keyNames = getKeyNames(json)
+    keyNames.contains(TYPE_FIELD)
+  }
+
+  def getJsonFirstLevelFieldsMap(json: Json): Map[String, Json] = {
+    for {
+      jobject <- json.asObject.toList
+      (key, value) <- jobject.toVector
+    } yield (key, value)
+  }.toMap
+
+  def getKeyNames(json: Json): List[String] = for {
+    jobject <- json.asObject.toList
+    k <- jobject.keys
+  } yield k
+
+  def resolveJsonTypingResult(json: Json): TypingResult = {
```

Comment:
WDYT about keeping code that prepares definition and implementaion (`TypedMapUtils`) in one class to avoid desynchronization of this code? Take a look at `InputMeta` for example. Also would be nice to use some important keyword in the name of this class. Like `Json` , `TypedMap` maybe just `JsonToTypedMapConverter`? `TypedMapUtils` doesn't saying too much. I know that someone can take a look at package name but I'm a big fun of using find by name IDE feature.

Reply:
Perhaps that is good idea, for now on I just updated `TypedMapUtils` to more meaningful `JsonToTypedMapConverter` as you suggested.

Reply:
It looks that it is not a good "in the future", but it is needed now - you already made those two places of code desynchronized changing scala BigDecimal to java BigDeicimal. Take a look at `SchemaTypingResult`